GameEngine/GameState.cs

```diff
@@ -50,9 +57,10 @@ public override bool Equals(object o)
             return other != null
                 && Board.Equals(other.Board)
                 && Deck.Equals(other.Deck)
-                && Hand.Equals(other.Hand)
+                && Hands.Count == other.Hands.Count && Hands.Zip( other.Hands, (a,b) => a.Equals(b) ).All( b => b )
                 && SunSpaces == other.SunSpaces
-                && SunCounter == other.SunCounter;
+                && SunCounter == other.SunCounter
+                && CurrentPlayer == other.CurrentPlayer;
```

Comment:
I have a slight preference for grouping the 'player-checks' together.  
I feel like testing the equality of the hand contents and the identity of the current player belong together.
I would tend to group them last since hand-equivalence is the most expensive part of the equality check.

Reply:
👍